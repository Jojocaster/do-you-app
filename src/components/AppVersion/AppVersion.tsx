import { Entypo } from '@expo/vector-icons'
import React from 'react'
import { MonoText } from '../StyledText'
import { View } from '../Themed'
import Constants from 'expo-constants'

export const AppVersion: React.FC = () => {
  const version = Constants.manifest?.version || ''
  return (
    <View
      style={{
        backgroundColor: 'rgba(255,255,255,0.05)',
        marginVertical: 20,
        padding: 10,
      }}
    >
      <MonoText>
        <Entypo name="code" size={15} /> App version: {version}
      </MonoText>
    </View>
  )
}
