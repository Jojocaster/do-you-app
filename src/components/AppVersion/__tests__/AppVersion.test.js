import * as React from 'react'
import renderer from 'react-test-renderer'

import { AppVersion } from '../AppVersion'
import Constants from 'expo-constants'
import { Text } from '../../Themed'

const mockVersion = '2.1.2'
jest.mock('expo-constants', () => ({
  manifest: {
    version: 'foo',
  },
}))

describe('AppVersion', () => {
  it('should render correctly', () => {
    const instance = renderer.create(<AppVersion />)
    expect(instance.toJSON()).toMatchSnapshot()
  })

  it('should render the correct app version', () => {
    Constants.manifest.version = mockVersion
    const instance = renderer.create(<AppVersion />).root
    expect(instance?.findByType(Text).props.children[2]).toEqual(mockVersion)
  })
})
