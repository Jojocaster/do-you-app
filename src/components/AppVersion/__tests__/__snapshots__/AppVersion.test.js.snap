// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppVersion should render correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#1c1c1c",
      },
      Object {
        "backgroundColor": "rgba(255,255,255,0.05)",
        "marginVertical": 20,
        "padding": 10,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#fff",
        },
        Array [
          undefined,
          Object {
            "fontFamily": "space-mono",
          },
        ],
      ]
    }
  >
    <Text />
     App version: 
    foo
  </Text>
</View>
`;
